source/renderer/app/App.js

```diff
@@ -14,25 +14,57 @@ import type { ActionsMap } from './actions/index';
 import ThemeManager from './ThemeManager';
 import AboutDialog from './containers/static/AboutDialog';
 import NetworkStatusDialog from './containers/status/NetworkStatusDialog';
+import GenericNotificationContainer from './containers/notifications/GenericNotificationContainer';
+import GenericNotification from './components/notifications/GenericNotification';
+import { DOWNLOAD_LOGS_SUCCESS_DURATION } from './config/timingConfig';
+import { i18nContext } from './utils/i18nContext';
+
+export const messages = defineMessages({
+  downloadLogsSuccess: {
+    id: 'loading.screen.reportIssue.downloadLogsSuccessMessage',
+    defaultMessage: '!!!Logs successfully downloaded',
+    description: 'Success message for download logs.',
+  },
+  downloadLogsProgress: {
+    id: 'loading.screen.reportIssue.downloadLogsProgressMessage',
+    defaultMessage: '!!!Preparing logs for download',
+    description: 'Progress message for download logs.',
+  },
+});
+
+const DOWNLOAD_LOGS_PROGRESS_NOTIFICATION_ID =
+  'loading-page-download-logs-progress';
+const DOWNLOAD_LOGS_SUCCESS_NOTIFICATION_ID =
+  'loading-page-download-logs-success';
 
 @observer
 export default class App extends Component<{
   stores: StoresMap,
   actions: ActionsMap,
   history: Object,
 }> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
   componentWillMount() {
     // loads app's global environment variables into AppStore via ipc
     this.props.actions.app.initAppEnvironment.trigger();
   }
   render() {
     const { stores, actions, history } = this.props;
     const { app } = stores;
-    const { isAboutDialogOpen, isNetworkStatusDialogOpen } = app;
+    const {
+      isAboutDialogOpen,
+      isNetworkStatusDialogOpen,
+      isNotificationVisible,
+    } = app;
     const locale = stores.profile.currentLocale;
+    const intl = i18nContext(locale);
     const mobxDevTools = global.environment.mobxDevTools ? <DevTools /> : null;
     const { currentTheme } = stores.profile;
     const themeVars = require(`./themes/daedalus/${currentTheme}.js`).default;
+    const notificationOrder = 99999;
```

Comment:
@DeeJayElly please add a comment regarding this order 🙏

Reply:
@nikolaglumac done